src/main.rs

```diff
@@ -462,7 +529,13 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
                 .collect::<Vec<_>>()
                 .join("/"),
             item_id,
-            recipe_id: recipe.id,
+            unknown_recipes: profitable_item.used_recipes.iter().filter_map(|(id, known)| {
+                if !*known {
+                    Some(*id)
+                } else {
+                    None
+                }
+            }).collect(),
```

Comment:
It can be convenient to dereference in the arguments:
```suggestion
            unknown_recipes: profitable_item.used_recipes.iter().filter_map(|(&id, &known)| {
                if !known {
                    Some(id)
                } else {
                    None
                }
            }).collect(),
```

Reply:
Good to learn. Resolved by using a HashSet.